WordPress/Classes/ViewRelated/Site Creation/Site Name/SiteNameViewController.swift

```diff
@@ -4,6 +4,10 @@ import UIKit
 /// Site Name screen for the Site Creation flow
 class SiteNameViewController: UIViewController {
 
+    private var nameView: SiteNameView {
+        return view as! SiteNameView
+    }
```

Comment:
This works, but it defeats the purpose of abstracting away `SiteNameView` from `SiteNameViewController` , thus, making the view not substitutable. For similar reasons, force unwrapping might be undesirable, because substituting the view would cause this to fail.

Another possible approach (thanks @diegoreymendez for helping with this!!) could be to customize the responder chain in `SiteNameView` and, in `SiteNameViewController`, only call `view.becomeFirstResponder()` inside `viewDidAppear(_:)`, or `viewWillAppear(_:)`, depending if we want to show the keyboard animation.

Something like:

in `SiteNameView`:

```
override var canBecomeFirstResponder: Bool {
        true
    }

    override func becomeFirstResponder() -> Bool {
        super.becomeFirstResponder()
        return searchBar.becomeFirstResponder()
    }
```

in `SiteNameViewController`:

```
override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        view.becomeFirstResponder()
    }
```
How does all this sound?

Reply:
>This works, but it defeats the purpose of abstracting away SiteNameView from SiteNameViewController , thus, making the view not substitutable. For similar reasons, force unwrapping might be undesirable, because substituting the view would cause this to fail.

IMO the `SiteNameViewController` will always have a direct relationship with `SiteNameView` since we're overriding `loadView`, and this is a pattern [others have used as well](https://swiftrocks.com/writing-cleaner-view-code-by-overriding-loadview). That said I do like the idea you've proposed so I'll implement that. 👍